matrix/buffer.py

```diff
@@ -703,24 +723,44 @@ def join(self, user, date, message=True, extra_tags=None):
             self.print_date_tags(msg, date, tags)
             self.add_smart_filtered_nick(user.nick)
 
-    def invite(self, nick, date, extra_tags=None):
+    def change_nick(self, user_id, old_nick, new_nick, date, message=True, extra_tags=None):
+        # type: (str, str, int, Optional[bool], Optional[List[str]]) -> None
+        user = self._get_user(user_id)
+        self._remove_user_from_nicklist(user)
+        user.nick = new_nick
+        self._add_user_to_nicklist(user)
+
+        if message:
+            tags = self._message_tags(user, "change")
+            msg = self._nick_change_message(user, old_nick, new_nick)
+
+            # TODO add a option to disable smart filters
+            tags.append(SCRIPT_NAME + "_smart_filter")
```

Comment:
The irc plugin adds two more tags for nick changes: _irc_nick1_ and _irc_nick2_.
Those tags contain the old and new nick and as far as i remember the smart filtering algorithm needs them to unmask lines that belong to the old nick.
More info [here](https://github.com/weechat/weechat/blob/master/src/plugins/irc/irc-protocol.c#L1486).

Reply:
interesting, so i guess i cannot use spaces or special symbols in tags there during irc_nick1_%s construction. Will those tags serve any purpose for weechat if I strip displayname of whitespace, unicode and/or special symbols?

Reply:
Oh right, display names can contain spaces.

I guess we'll have to change the smart filter algorithm to just use the user_id, and put the user_id into the tags of every message.